src/tools/rustfmt/Cargo.toml

```diff
@@ -37,7 +37,7 @@ annotate-snippets = { version = "0.9", features = ["color"] }
 anyhow = "1.0"
 bytecount = "0.6"
 cargo_metadata = "0.14"
-clap = { version = "3.1", features = ["derive"] }
+clap = { version = "4.2.1", features = ["derive"] }
```

Comment:
@rust-lang/rustfmt Are you OK with this landing upstream? It seems like there's some changes to code here as well so maybe we should go through PR to rustfmt?

Reply:
Adding clap 3.1 gave us some trouble https://github.com/rust-lang/rustfmt/issues/5395, but I think bumping to 4.2.1 upstream should be fine. I've gone ahead and tested the 4.2.1 changes locally. There are some slight changes in the `--help` output, but I also think that's fine. @calebcartwright what are your thoughts?

## 3.1
<img width="682" alt="Screen Shot 2023-04-09 at 12 43 26 PM" src="https://user-images.githubusercontent.com/29028348/230785372-4063b433-c056-478e-bdbf-9f13796813a0.png">

## 4.2.1
<img width="673" alt="Screen Shot 2023-04-09 at 12 45 03 PM" src="https://user-images.githubusercontent.com/29028348/230785407-45e0f57d-30ae-45a2-b198-8b8d29408351.png">

Reply:
In general I don't think subtrees should be updated in tree unless absolutely necessary (e.g. rustc_ast updates requiring corresponding updates in clippy, rustfmt, etc.). I understand the benefit of making a broad update like this in tree, but I don't particularly like it (we run a reduced test suite here, lockfile considerations can complicate subtree syncs, etc.)

Suppose we can let this one slide (thanks Yacin for taking a closer look), but in the future would really want to see something like this come directly to the rustfmt repo and then make its way in-tree here via the usual sync process